java/vector/src/main/java/org/apache/arrow/vector/file/ArrowReader.java

```diff
@@ -17,227 +17,175 @@
  */
 package org.apache.arrow.vector.file;
 
-import com.google.common.collect.Iterators;
-import io.netty.buffer.ArrowBuf;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.vector.FieldVector;
-import org.apache.arrow.vector.complex.DictionaryVector;
+import org.apache.arrow.vector.VectorLoader;
+import org.apache.arrow.vector.VectorSchemaRoot;
 import org.apache.arrow.vector.schema.ArrowDictionaryBatch;
-import org.apache.arrow.vector.schema.ArrowFieldNode;
 import org.apache.arrow.vector.schema.ArrowMessage;
 import org.apache.arrow.vector.schema.ArrowMessage.ArrowMessageVisitor;
 import org.apache.arrow.vector.schema.ArrowRecordBatch;
-import org.apache.arrow.vector.schema.VectorLayout;
-import org.apache.arrow.vector.types.Dictionary;
-import org.apache.arrow.vector.types.Types;
-import org.apache.arrow.vector.types.Types.MinorType;
+import org.apache.arrow.vector.dictionary.Dictionary;
 import org.apache.arrow.vector.types.pojo.ArrowType;
 import org.apache.arrow.vector.types.pojo.DictionaryEncoding;
 import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.types.pojo.Schema;
 
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-
-import static com.google.common.base.Preconditions.checkArgument;
-
-public abstract class ArrowReader<T extends ReadChannel> implements AutoCloseable {
-
-    private final T in;
-    private final BufferAllocator allocator;
-    private Schema schema;
-
-    private List<FieldVector> vectors;
-    private Map<String, FieldVector> vectorsByName;
-    private Map<Long, FieldVector> dictionaries;
-
-    private int batchCount = 0;
-    private boolean initialized = false;
-
-    protected ArrowReader(T in, BufferAllocator allocator) {
-        this.in = in;
-        this.allocator = allocator;
+public abstract class ArrowReader<T extends ReadChannel> extends ArrowMagic implements AutoCloseable {
```

Comment:
not following... ArrowMagic is an interface, the MAGIC variable is static but not imported since it's in scope already.